pkg/render/whisker/component.go

```diff
@@ -0,0 +1,259 @@
+	return c
+}
+
+// Configuration contains all the config information needed to render the component.
+type Configuration struct {
+	PullSecrets                 []*corev1.Secret
+	OpenShift                   bool
+	Installation                *operatorv1.InstallationSpec
+	TunnelSecret                *corev1.Secret
+	TrustedCertBundle           certificatemanagement.TrustedBundle
+	LinseedPublicCASecret       *corev1.Secret
+	ManagementClusterConnection *operatorv1.ManagementClusterConnection
+}
+
+type WhiskerComponent struct {
+	cfg   *Configuration
+	dep   *_k8s.Deployment
+	image string
+}
+
+func (c *WhiskerComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	return c.dep.ApplyImageSet(c.cfg.Installation, is)
+}
+
+func (c *WhiskerComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *WhiskerComponent) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		render.CreateNamespace(WhiskerNamespace, c.cfg.Installation.KubernetesProvider, render.PSSRestricted, c.cfg.Installation.Azure),
+	}
+
+	objs = append(objs,
+		render.CreateOperatorSecretsRoleBinding(WhiskerNamespace),
+		c.serviceAccount(),
+		c.clusterRole(),
+		c.clusterRoleBinding())
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(WhiskerNamespace, c.cfg.PullSecrets...)...)...)
+
+	objs = append(objs, c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace))
+	objs = append(objs, c.dep.Objects()...)
+
+	if c.cfg.ManagementClusterConnection != nil && c.cfg.TunnelSecret != nil {
+		objs = append(objs, secret.CopyToNamespace(WhiskerNamespace, c.cfg.TunnelSecret)[0])
+	}
+
+	return objs, nil
+}
+
+func (c *WhiskerComponent) Ready() bool {
+	return true
+}
+
+func (c *WhiskerComponent) serviceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: WhiskerServiceAccountName, Namespace: WhiskerNamespace},
+	}
+}
+
+func (c *WhiskerComponent) whisker() *_k8s.Container {
+	return _k8s.NewContainer("whisker", components.Component{
+		Image:    "calico/whisker",
+		Version:  "bmv1.10",
+		Registry: "gcr.io/unique-caldron-775/brianmcmahon/",
+	}).AddEnv(
+		[]corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "DEBUG"},
+			{Name: "CA_CERT_PATH", Value: c.cfg.TrustedCertBundle.MountPath()},
+		}...).MountConfigMap(c.cfg.TrustedCertBundle.VolumeMountPath(rmeta.OSTypeLinux), c.cfg.TrustedCertBundle.ConfigMap("")).
+		AddService(&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{Name: "whisker"},
+			Spec:       corev1.ServiceSpec{Ports: []corev1.ServicePort{{Port: 8081}}},
+		})
+}
+
+func (c *WhiskerComponent) whiskerBackend() *_k8s.Container {
+	return _k8s.NewContainer("whisker-backend", components.Component{
+		Image:    "calico/whisker-backend",
+		Version:  "bmv1.10",
+		Registry: "gcr.io/unique-caldron-775/brianmcmahon/",
+	}).AddEnv([]corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "DEBUG"},
```

Comment:
probably shouldn't default to DEBUG in production - do we need a config knob for this?